### ISSUE TITLE
Indentation bug?

### ISSUE THREAD

**Issue by MEO265** [r-lib/lintr#2144]
I think the `indentation_linter()` has a bug. The following code gives me the error message below on line 6. 
```r
test_that("unreachable_code_linter works in sub expressions", {
  linter <- unreachable_code_linter()
  msg <- rex::rex("Code and comments coming after a return() or stop()")

  expect_lint(
    lines,
    list(
      list(line_number = 4L, message = msg),
      list(line_number = 7L, message = msg),
      list(line_number = 10L, message = msg),
      list(line_number = 15L, message = msg)
    ),
    linter)
})
```
`Hanging indent should be 14 spaces but is 4 spaces.`

See also PR #2141

**Comment by MEO265** [comment]
I see my mistake, which is that I forgot the line break before the last `)`, but the error message is more confusing than helpful.

It would really look like this:
```r
expect_lint(
    lines,
    list(
      list(line_number = 4L, message = msg),
      list(line_number = 7L, message = msg),
      list(line_number = 10L, message = msg),
      list(line_number = 15L, message = msg)
    ),
    linter
  )
```

Since this is not a common error, you should receive a separate message in the right place.

**Comment by MichaelChirico** [comment]
Unfortunately it's not so simple, because this would also be valid:

```r
test_that("unreachable_code_linter works in sub expressions", {
  linter <- unreachable_code_linter()
  msg <- rex::rex("Code and comments coming after a return() or stop()")

  expect_lint(
              lines,
              list(
                list(line_number = 4L, message = msg),
                list(line_number = 7L, message = msg),
                list(line_number = 10L, message = msg),
                list(line_number = 15L, message = msg)
              ),
              linter)
})
```

The trailing `)` points the lint message to this code being in `"hanging"` format.

The lint message could probably be more helpful, but it's not wrong.

**Comment by AshesITR** [comment]
A hanging indent without an argument on the opening line certainly does seem weird.
Maybe we should constrain hanging indents to only apply if both parentheses have a token on the same line.

In that case there should still be a lint on the closing paren of a block indent, stating that it should be on its own line.

Special casing for `test_that("bla", { ... })` will and friends probably necessary.

**Comment by d-morrison** [comment]
I've been running into this issue too; for example:

``` r
code <-
'
lm(
  data = iris,
  formula = Sepal.Length ~ Sepal.Width)
'

lintr::lint(code)
#> <text>:3:2: style: [indentation_linter] Hanging indent should be 3 spaces but is 2 spaces.
#>   data = iris,
#>  ^~
```

<sup>Created on 2025-03-11 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

If I'm understanding the style guide correctly, the issue here is actually that the closing `)` needs to be moved to its own line.

I can't tell from https://style.tidyverse.org/syntax.html#long-function-calls whether hanging-indent is allowed for function *calls*, but if so, an alternative correction of the correction of the code above would be to add a third space AND move `data = iris` onto the first line; i.e.,

``` r
lm(data = iris,
   formula = Sepal.Length ~ Sepal.Width)
```

In other words, even if hanging indents are allowed, shouldn't the following code throw a lint due to the first argument not appearing on the initial line?

``` r
code1 <-
'
lm(
   data = iris,
   formula = Sepal.Length ~ Sepal.Width)
'

lintr::lint(code1)
#> ℹ No lints found.
```

<sup>Created on 2025-03-11 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>
